main.tf

```diff
@@ -29,37 +55,53 @@ resource "aws_launch_configuration" "this" {
 # Autoscaling group
 ####################
 resource "aws_autoscaling_group" "this" {
-  count = "${var.create_asg && !var.create_asg_with_initial_lifecycle_hook ? 1 : 0}"
-
-  name_prefix          = "${join("-", compact(list(coalesce(var.asg_name, var.name), var.recreate_asg_when_lc_changes ? element(concat(random_pet.asg_name.*.id, list("")), 0) : "")))}-"
-  launch_configuration = "${var.create_lc ? element(concat(aws_launch_configuration.this.*.name, list("")), 0) : var.launch_configuration}"
-  vpc_zone_identifier  = ["${var.vpc_zone_identifier}"]
-  max_size             = "${var.max_size}"
-  min_size             = "${var.min_size}"
-  desired_capacity     = "${var.desired_capacity}"
-
-  load_balancers            = ["${var.load_balancers}"]
-  health_check_grace_period = "${var.health_check_grace_period}"
-  health_check_type         = "${var.health_check_type}"
-
-  min_elb_capacity          = "${var.min_elb_capacity}"
-  wait_for_elb_capacity     = "${var.wait_for_elb_capacity}"
-  target_group_arns         = ["${var.target_group_arns}"]
-  default_cooldown          = "${var.default_cooldown}"
-  force_delete              = "${var.force_delete}"
-  termination_policies      = "${var.termination_policies}"
-  suspended_processes       = "${var.suspended_processes}"
-  placement_group           = "${var.placement_group}"
-  enabled_metrics           = ["${var.enabled_metrics}"]
-  metrics_granularity       = "${var.metrics_granularity}"
-  wait_for_capacity_timeout = "${var.wait_for_capacity_timeout}"
-  protect_from_scale_in     = "${var.protect_from_scale_in}"
-
-  tags = ["${concat(
-      list(map("key", "Name", "value", var.name, "propagate_at_launch", true)),
+  count = var.create_asg && false == var.create_asg_with_initial_lifecycle_hook ? 1 : 0
+
+  name_prefix = "${join(
+    "-",
+    compact(
+      [
+        coalesce(var.asg_name, var.name),
+        var.recreate_asg_when_lc_changes ? element(concat(random_pet.asg_name.*.id, [""]), 0) : "",
+      ],
+    ),
+  )}-"
+  launch_configuration = var.create_lc ? element(concat(aws_launch_configuration.this.*.name, [""]), 0) : var.launch_configuration
+  vpc_zone_identifier  = var.vpc_zone_identifier
+  max_size             = var.max_size
+  min_size             = var.min_size
+  desired_capacity     = var.desired_capacity
+
+  load_balancers            = var.load_balancers
+  health_check_grace_period = var.health_check_grace_period
+  health_check_type         = var.health_check_type
+
+  min_elb_capacity          = var.min_elb_capacity
+  wait_for_elb_capacity     = var.wait_for_elb_capacity
```

Comment:
also this attribute is a defined as a boolean at variables.tf (=false) but [wait_for_elb_capacity] in the  aws_autoscaling group resource awaiting for [int]

Reply:
Looks like `var.wait_for_elb_capacity` should be a number, so I've set that type in the variables.tf.. I also set the default here to `null`, because otherwise it will always override settings for `var.min_elb_capacity`